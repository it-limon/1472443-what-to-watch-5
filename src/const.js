export const MAX_COMMENTS_PER_COLUMN = 2;
export const MAX_GENRES_COUNT = 9;
export const SIMILAR_MOVIES_COUNT = 4;
export const DEFAULT_SHOWN_MOVIES_COUNT = 8;
export const DEFAULT_REVIEW_RATING = 3;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const ALL_GENRES = `All genres`;

export const MovieRating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppPages = {
  MAIN: `MAIN`,
  AUTH: `AUTH`,
  MOVIE: `MOVIE`,
  MYLIST: `MYLIST`,
  REVIEW: `REVIEW`,
  PLAYER: `PLAYER`
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FILMS: `/films`,
  MYLIST: `/mylist`,
  REVIEW: `/review`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  FAVORITE_FILMS: `/favorite`,
  PROMO_MOVIE: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`
};
