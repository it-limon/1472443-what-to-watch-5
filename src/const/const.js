export const MovieRatings = [`Bad`, `Normal`, `Good`, `Very Good`, `Awesome`];
export const REVIEW_COLUMNS_COUNT = 2;
export const LIKE_THIS_MOVIE_COUNT = 4;
