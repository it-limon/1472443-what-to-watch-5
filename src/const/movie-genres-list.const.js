export const MovieGenresList = [
  {
    key: 0,
    ord: 0,
    name: `All genres`,
    filterName: `All genres`,
  },
  {
    key: 1,
    ord: 1,
    name: `Comedie`,
    filterName: `Comedies`,
  },
  {
    key: 2,
    ord: 2,
    name: `Crime`,
    filterName: `Crime`,
  },
  {
    key: 3,
    ord: 3,
    name: `Documentary`,
    filterName: `Documentary`,
  },
  {
    key: 4,
    ord: 4,
    name: `Drama`,
    filterName: `Dramas`,
  },
  {
    key: 5,
    ord: 5,
    name: `Horror`,
    filterName: `Horror`,
  },
  {
    key: 6,
    ord: 6,
    name: `Kids & Family`,
    filterName: `Kids & Family`,
  },
  {
    key: 7,
    ord: 7,
    name: `Romance`,
    filterName: `Romance`,
  },
  {
    key: 8,
    ord: 8,
    name: `Sci-Fi`,
    filterName: `Sci-Fi`,
  },
  {
    key: 9,
    ord: 9,
    name: `Thriller`,
    filterName: `Thrillers`,
  }
];
