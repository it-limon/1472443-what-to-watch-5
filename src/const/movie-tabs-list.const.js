export const MovieTabsList = [
  {
    key: 0,
    name: `Overview`
  },
  {
    key: 1,
    name: `Details`
  },
  {
    key: 2,
    name: `Reviews`
  }
];
