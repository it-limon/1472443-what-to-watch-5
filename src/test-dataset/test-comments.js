export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];