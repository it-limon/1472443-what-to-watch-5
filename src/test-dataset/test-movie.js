export const testMovie = {
  id: 1,
  name: `El Camino: A Breaking Bad Movie`,
  posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
  previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
  backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
  backgroundColor: `#000000`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
  rating: 7.2,
  scoresCount: 36346,
  director: `Vince Gilligan`,
  starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
  runTime: 122,
  genre: `Drama`,
  released: 2019,
  isFavorite: true
};
