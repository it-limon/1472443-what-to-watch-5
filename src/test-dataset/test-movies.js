export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];