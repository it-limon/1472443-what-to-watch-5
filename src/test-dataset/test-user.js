export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};