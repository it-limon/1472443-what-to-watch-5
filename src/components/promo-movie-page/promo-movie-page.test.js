import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import PromoMoviePage from "./promo-movie-page";
import {Provider} from "react-redux";
import {Route, BrowserRouter} from "react-router-dom";
import {AuthorizationStatus} from "../../const";
import {testMovie} from "../../test-dataset/test-movie";
import {testUser} from "../../test-dataset/test-user";

let mockStore = null;
let store = null;

describe(`Render PromoMoviePage`, () => {

  it(`Render PromoMoviePage (no auth)`, () => {
    mockStore = configureStore([]);
    store = mockStore({
      DATA: {
        promoMovie: testMovie
      },
      STATE: {
        lastActiveMovie: testMovie
      },
      USER: {
        authorizationStatus: AuthorizationStatus.NO_AUTH
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <PromoMoviePage />
            </Route>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render PromoMoviePage (auth)`, () => {
    mockStore = configureStore([]);
    store = mockStore({
      DATA: {
        promoMovie: testMovie
      },
      STATE: {
        lastActiveMovie: testMovie
      },
      USER: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userInfo: testUser
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <PromoMoviePage />
            </Route>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
