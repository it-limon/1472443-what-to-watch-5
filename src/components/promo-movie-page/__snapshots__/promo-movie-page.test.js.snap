// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PromoMoviePage Render PromoMoviePage (auth) 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="El Camino: A Breaking Bad Movie"
      src="https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <div
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </div>
    </div>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User Name"
            height="63"
            src="https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="El Camino: A Breaking Bad Movie poster"
          height="327"
          src="https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          El Camino: A Breaking Bad Movie
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2019
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="14"
              viewBox="0 0 18 14"
              width="18"
            >
              <use
                xlinkHref="#in-list"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Render PromoMoviePage Render PromoMoviePage (no auth) 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="El Camino: A Breaking Bad Movie"
      src="https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <div
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </div>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="El Camino: A Breaking Bad Movie poster"
          height="327"
          src="https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          El Camino: A Breaking Bad Movie
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2019
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
